### PR TITLE
Polish erlang module docs

### PR DESCRIPTION
**Pull request by garazdawi**
Create sections for all functions in the Erlang module so that they can more easily be found when exploring.

### CODE REVIEW

**Review by kikofernandez — APPROVED** [review]
This looks like a good way to organise functions. I think it is also easier to find in the left-hand side panel the information by looking at Checksum, Types, Erlang Terms, etc.
Looks great!

**Review by rickard-green — APPROVED** [review]
Nice!

File: erts/preloaded/src/erlang.erl [hunk 1]
```diff
@@ -21,28 +21,22 @@
...
+in this module. Some of the BIFs and all of the predefined types are viewed more
+or less as part of the Erlang programming language and are _auto-imported_.
+Thus, it is not necessary to specify the module name. For example, the calls
+[`atom_to_list(erlang)`](`atom_to_list/1`) and [`erlang:atom_to_list(erlang)`](`atom_to_list/1`)
+are identical.
+
+Auto-imported BIFs are annotated with `auto-imported` and predefined types are
+annotated with `predefined`.
+
+Some auto-imported BIFs are also allowed in [guard expression](`e:system:expressions.md#guard-expressions`). Such BIFs are annoted with both `auto-imported` and `guard-bif`.
```
Comment (bjorng) [inline erts/preloaded/src/erlang.erl:35]:
```suggestion
Some auto-imported BIFs are also allowed in [guard expression](`e:system:expressions.md#guard-expressions`).
Such BIFs are annoted with both `auto-imported` and `guard-bif`.
```

File: erts/preloaded/src/erlang.erl [hunk 2]
```diff
@@ -2033,8 +2089,9 @@ alias(_Opts) ->
     erlang:nif_error(undefined).
 
 -doc """
-Deactivate the alias `Alias` previously created by the calling process. An alias
-can for example be created via `alias/0`, or `monitor/3`.
+Deactivate the alias `Alias` previously created by the calling process.
+
+An alias can for example be created via `alias/0`, or `monitor/3`.
```
Comment (bjorng) [inline erts/preloaded/src/erlang.erl:2094]:
```suggestion
An alias can, for example, be created via `alias/0` or `monitor/3`.
```

File: erts/preloaded/src/erlang.erl [hunk 3]
```diff
@@ -11168,11 +11506,13 @@ nodes(_Arg) ->
...
 -doc """
-Returns a list of `NodeInfo` tuples. The first element is the node name. Nodes
-to be included in the list are determined by the first argument `Arg` in the
-same way as for [`nodes(Arg)`](`nodes/1`). The second element of `NodeInfo`
-tuples is a map containing further information about the node identified by the
-first element. The information present in this map is determined by the
+Returns a list of `NodeInfo` tuples.
+
+The first element is the node name. Nodes to be included in the list are determined
+ by the first argument `Arg` in the same way as for [`nodes(Arg)`](`nodes/1`).
```
Comment (bjorng) [inline erts/preloaded/src/erlang.erl:11512]:
```suggestion
by the first argument `Arg` in the same way as for [`nodes(Arg)`](`nodes/1`).
```

**Review by bjorng — APPROVED** [review]
Nice! I only have a few nitpicks.

**Comment by Maria-12648430** [comment]
I'm a bit late to the party here, sorry :sweat_smile:

This looks pretty nice, but I just noticed that I have trouble finding specific functions. That is, I know the module and name of the function I'm looking for, but I'm not sure about the category where I can find it.

It would be great if, aside from the categorized index, there would also be an index of _all_ the functions in a module. This categorization is nice if I was looking for functions to perform a specific task or discover what a module has to offer in that regard, checksumming for example, but not so good if I have a function and want to find details about it.

Like, most of the time, I know _exactly_ the function I want to look up, ie I'm just unsure about something like the order of arguments or some other detail. In the old documentation, I would just open up the module docs, scroll down the alphabetical index to the function, and there it was. Now, I have to figure out what category it might best fit in, or rather, in which category _the person writing the documentation_ thought it would best fit in. Take `phash2` for example. It's in the "Erlang Terms" category (which is ok), but "Checksums" would also be fitting.

**Comment by garazdawi** [comment]
If you know what you are looking for, why not search for it? Either just using the built-in brower search, or the ExDoc search?

**Comment by Maria-12648430** [comment]
That _would_ be an option, yes :sweat_smile: TBH, the thought never occurred to me, probably because I never used the search in the old documentation. Still, I think an index of _all_ functions (in the nav bar) would be handy. I guess this is an issue for `ex_doc`, however.

**Comment by garazdawi** [comment]
> That would be an option, yes 😅 TBH, the thought never occurred to me, probably because I never used the search in the old documentation.

I used to click around as well, but since the addition of the algolia search it is much simpler to just hit "Ctrl-k" and then type `lists:foreach`.

> Still, I think an index of all functions (in the nav bar) would be handy. I guess this is an issue for ex_doc, however.

It is not something that is possible to do with ExDoc at the moment. A function/type is either in a group, or it is not. So yes, it would be an extension to ExDoc.

**Pull request merged**